### PR TITLE
Update dependencies and jdk

### PR DESCRIPTION
**Pull request by hiroshitoda**

### CODE REVIEW

**Comment by NozomiIto** [comment]
Thanks so much for your contribution!
But I'm afraid Travis build seems to fail.

**Comment by hiroshitoda** [comment]
uh, geckodriver is needed... sorry.
I fixed and pushed them.

and, would you like to remain Java7 settings on Travis?
if Selenium 3, I think Java7 build target should be removed.

**Comment by hiroshitoda** [comment]
```
java.lang.IllegalStateException: The driver is not executable: /home/travis/build/SahaginOrg/sahagin-java/src/test/resources/selenium/linux/geckodriver
	at org.sahagin.report.HtmlReportTest.seleniumTestRun(HtmlReportTest.java:126)
	at org.sahagin.report.HtmlReportTest.generatedReportShouldWorkOnFirefox(HtmlReportTest.java:147)
```
I commited [geckodriver-v0.18.0-linux64.tar.gz](https://github.com/mozilla/geckodriver/releases/download/v0.18.0/geckodriver-v0.18.0-linux64.tar.gz) .
I think it's suitable for Travis. why can't we use it on Travis?

**Comment by NozomiIto** [comment]
> I think it's suitable for Travis. why can't we use it on Travis?

Yes, good!

> and, would you like to remain Java7 settings on Travis?
> if Selenium 3, I think Java7 build target should be removed.

Yeah, now we should use Java 1.8 + Selenium3 in sahagin-java/pom.xml, and since Selenium3 requires Java8, we don't need to test Sahagin for Java8 any more.

And for your information, I'd like to refrain from using Java8 specific syntax as long as possible so that our users can use the latest Sahagin with old Selenium and Java1.7 by tweaking their pom.xml!

**Comment by hiroshitoda** [comment]
> And for your information, I'd like to refrain from using Java8 specific syntax as long as possible so that our users can use the latest Sahagin with old Selenium and Java1.7 by tweaking their pom.xml!

I should modify dependencies for JDK7/8 by pom.xml profiles. please wait for my additional commits.

**Comment by NozomiIto** [comment]
> I should modify dependencies for JDK7/8 by pom.xml profiles. please wait for my additional commits.

I think Sahagin itself's pom.xml only has to support Java8 dependency, and your modified pom.xml seems already good enough. 
If some user wants to use the next latest Sahagin with Java7 or old Selenium, they should override Java8 and Selenium3 dependencies by changing their test project's pom.xml.

But is there any other problem on your changed pom.xml?

**Comment by hiroshitoda** [comment]
I made profiles Java7 and Java8. Java8 profile is default.
Java7 users can switch by only profile specifying. How about this?

I wonder Travis can't build this project properly...(I can execute this on Windows with no error.)

**Comment by NozomiIto** [comment]
> I made profiles Java7 and Java8. Java8 profile is default.
> Java7 users can switch by only profile specifying. How about this?

Sounds very nice!

**Comment by NozomiIto** [comment]
> I wonder Travis can't build this project properly...(I can execute this on Windows with no error.)

https://travis-ci.org/SahaginOrg/sahagin-java/jobs/264336277#L1609

```
org.openqa.selenium.WebDriverException: 
Missing 'marionetteProtocol' field in handshake
```

https://stackoverflow.com/questions/38991440/webdriverexception-message-missing-marionetteprotocol-field-in-handshake

**Comment by NozomiIto** [comment]
What version of Firefox do you think TravisCI use?
For java8 build, I'm afraid Firefox version on TravisCI is too old.
travis.yaml seems to have Firefox version setting ( https://docs.travis-ci.com/user/firefox ), so I feel it would be better to set enough latest Firefox version.
As for java7 TravisCI build, you may only drop Travis CI build setting. 
Even if the user still can use Sahagin with Java7, we don't need to test the behavior for Java7. Java7 support is just best effort support.

**Comment by hiroshitoda** [comment]
geckodriver 0.18 is suitable for Firefox 53, I made some settings for that.
but, only in case of Firefox, NoSuchElement exception occured on Travis CI and Circle CI as same...
Chromium passed test.

**Comment by NozomiIto** [comment]
> but, only in case of Firefox, NoSuchElement exception occurred on Travis CI and Circle CI as same...

I tried your PR on my local PC and found NoSuchElement failure can occur also on my PC. This seems FirefoxDriver timing problem. FirefoxDriver seems to execute `driver.navigate().back()` before completing page switch. You can avoid this problem by fixing [this failed test lines](https://github.com/SahaginOrg/sahagin-java/blob/master/src/test/java/org/sahagin/report/HtmlReportTest.java#L119) like the following:

```
driver.findElement(By.linkText("sample.SampleTest.shouldSucceed")).click();
new WebDriverWait(driver, 60).until(ExpectedConditions.titleIs("shouldSucceed"));
driver.navigate().back();
```

Besides, committed geckodriver executable does not have the execute permission (and Maven test failed on my PC with Eclipse). You can add the permission to the file in Git repository with [this step](http://qiita.com/satotka/items/29f1483f8921d2ecfeab). By doing so, you don't need to call chmod command in circle.yml or travis.yml.

**Comment by hiroshitoda** [comment]
all green!

File: src/main/java/org/sahagin/runlib/external/adapter/fluentlenium/FluentLeniumAdapter.java [hunk 1]
```diff
@@ -26,16 +26,16 @@ public String getName() {
     }
 
     // can set null
-    public static void setAdapter(Fluent fluent) {
+    public static void setAdapter(FluentDriver fluent) {
         AdapterContainer container = AdapterContainer.globalInstance();
```
Comment (NozomiIto) [inline src/main/java/org/sahagin/runlib/external/adapter/fluentlenium/FluentLeniumAdapter.java:30]: @hiroshitoda Can this  break backward compatibility? If so, I will notice about this in the next release note.

**Comment by NozomiIto** [comment]
Merged. Thank you!!

File: src/main/java/org/sahagin/runlib/external/adapter/fluentlenium/FluentLeniumAdapter.java (hunk 1 above)
Comment (hiroshitoda) [reply src/main/java/org/sahagin/runlib/external/adapter/fluentlenium/FluentLeniumAdapter.java:30]: yes.
the architecture of FluentLenium was drastically changed from v.0.13.x to v.1.x.x.
`Fluent` on v.0.13.x is proxy of WebDriver. so I think `FluentDriver` on v.1.x.x is suitable for substitute.
this breaks backward compatibility.
please notify about that on changelog.

**Pull request merged**